sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java

```diff
@@ -98,7 +99,12 @@ public ConsistencyWriter(
         TimeoutHelper timeout,
         boolean forceRefresh) {
 
-        if (timeout.isElapsed()) {
+        if (timeout.isElapsed() &&
+            // skip throwing RequestTimeout on first retry because the first retry with
+            // force address refresh header can be critical to recover for example from
+            // stale Gateway caches etc.
+            entity.requestContext.retryContext != null &&
+            entity.requestContext.retryContext.retryCount > 1) {
```

Comment:
Currently `retryContext` doesn't work for query operations. I am not sure, but I am guessing query code path doesn't go through this class. Because if it does, then `retryContext` will be null, and this if block will not be executed.

Reply:
This is ConsistencyWriter = only ever be used for retry operations - never queries

Reply:
Sounds good.